src/apscheduler/datastores/mongodb.py

```diff
@@ -684,10 +686,16 @@ async def acquire_jobs(
                         continue
 
                     # Skip and un-acquire the job if no more slots are available
-                    if not task_job_slots_left.get(job.task_id, float("inf")):
+                    task_slots_left = task_job_slots_left.get(job.task_id, float("inf"))
+                    if (
+                        not task_slots_left
+                        or running_job_count_increments[job.task_id] == task_slots_left
+                    ):
```

Comment:
I think this would be the correct fix?
```suggestion
                    if task_job_slots_left.get(job.task_id, float("inf")):
```

Reply:
The problem with my current approach with MongoDB is the lack of atomicity. Other schedulers might change the task's running jobs count while this scheduler is running `acquire_jobs()`, as there is no locking involved. I think a better way to do this is to run a conditional update that only updates the running job count if it's lower than the maximum job count. My initial attempt at this failed because `find_and_update()`, for whatever bizarre reason, doesn't support referencing existing fields in the filter. But the next better way would be to use the previously fetched maximum running job counts, as those rarely change.

Reply:
> From what I understand, your change to
> 
> ```python
> if task_job_slots_left.get(job.task_id, float("inf")):
> ```
> 
> Would cause all tasks with a positive value, such as: 1, 2, 3 etc..., to be skipped.
> 
> Am I right or not?

You're right – I was admittely tired when writing that reply. But I think that part of my original logic was correct (`if not task_job_slots_left.get(job.task_id, float("inf")):`). What was wrong was my calculation of free slots (`task_job_slots_left[doc["_id"]] = doc["max_running_jobs"]`), as I should've deducted the number of slots in use from the max slots.